libs/blocks/global-navigation/blocks/profile/profile.js

```diff
@@ -9,87 +10,110 @@ function decorateEmail(email) {
-  const profileDetails = createTag('div', { class: 'gnav-profile-details' });
-  const profileActions = createTag('ul', { class: 'gnav-profile-actions' });
-
-  profileHeader.href = decorateProfileLink(accountLink.href, 'account');
-  profileHeader.setAttribute('aria-label', accountLink.textContent);
+};
 
-  const profileImg = avatarImg.cloneNode(true);
-  const profileName = createTag('p', { class: 'gnav-profile-name' }, displayName);
-  const profileEmail = createTag('p', { class: 'gnav-profile-email' }, displayEmail);
-  const accountText = blockEl.querySelector('div > div > p:nth-child(2) a').innerHTML;
-  const profileViewAccount = createTag('p', { class: 'gnav-profile-account' }, accountText);
-  profileDetails.append(profileName, profileEmail, profileViewAccount);
+const decorateAction = (actionEl) => {
+  if (!actionEl) return '';
+  actionEl.href = decorateProfileLink(actionEl.href, 'adminconsole');
+  return toFragment`<li class="feds-profile-action">${actionEl}</li>`;
+};
 
-  if (sections.manage.items.team?.id) {
-    const teamLink = blockEl.querySelector('div > div > p:nth-child(3) a');
-    teamLink.href = decorateProfileLink(teamLink.href, 'adminconsole');
-    const manageTeam = createTag('li', { class: 'gnav-profile-action' }, teamLink);
-    profileActions.append(manageTeam);
+class Profile {
+  constructor({
+    decoratedEl,
+    avatarImgEl,
+    sections,
+    toggleMenu,
+    profileButtonEl,
+    accountLinkEl,
+    signOutEl,
+    manageTeamsEl,
+    manageEnterpriseEl,
+    localMenu,
+  }) {
+    this.sections = sections;
+    this.avatarImgEl = avatarImgEl;
+    this.accountLinkEl = accountLinkEl;
+    this.signOutEl = signOutEl;
+    this.manageTeamsEl = manageTeamsEl;
+    this.manageEnterpriseEl = manageEnterpriseEl;
+    this.toggleMenu = toggleMenu;
+    this.profileButtonEl = profileButtonEl;
+    this.decoratedEl = decoratedEl;
+    this.localMenu = localMenu;
+    if (localMenu) {
+      localMenu.classList.add('feds-local-menu');
+    }
+    this.init();
   }
 
-  if (sections.manage.items.enterprise?.id) {
-    const manageLink = blockEl.querySelector('div > div > p:nth-child(4) a');
-    manageLink.href = decorateProfileLink(manageLink.href, 'adminconsole');
-    const manageEnt = createTag('li', { class: 'gnav-profile-action' }, manageLink);
-    profileActions.append(manageEnt);
+  async init() {
+    // TODO do some sanity checks if the user is logged in, the mandatory properties are set.
+    // If not there should be helpful logs providing guidance for developers
+    const { displayName, email } = await window.adobeIMS.getProfile();
+    if (this.profileButtonEl) this.profileButtonEl.setAttribute('aria-label', displayName);
+    this.displayName = displayName;
+    this.email = email;
+    this.decoratedEl.append(this.menu());
+    this.decoratedEl.addEventListener('click', () => this.toggleMenu(this.decoratedEl));
+    this.decoratedEl.dispatchEvent(new Event('profile_ready'));
```

Comment:
Do we have a use-case for this event in the new integration? If not, maybe it can be removed.

Reply:
Yeah, we use it to forward a click on the profile button, we can't really say when the profile is really initiated otherwise.

Reply:
Then may I suggest we come up with an event naming schema? Maybe `feds:events:profileReady` or something like that. I think `profile_ready` in particular was used due to Beagle historical context, but now I'd say we're free to switch it up. I prefer the more explicit naming due to the countless possible moving pieces on our pages. Maybe IMS also has an event when the Profile data is ready? Maybe an event is triggered when UAP data is available? I'd rather avoid collision and make things specific from the start.

Doesn't necessarily need to be part of this PR, just something to keep in mind.

Reply:
Hmm, I don't have a good picture yet on the backwards combability with the old profile. Or if we even want to be backwards compatible with the old probile. For now I went with `feds:events:profileReady` and we can still think about what/if we want to support in regards to the beagle profile 👍

Reply:
we don't need to have 100% backwards compatibility, just make sure we document changes properly.